HerPublicWebsite/Filters/SessionExpiryAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Linq;
+using HerPublicWebsite.Controllers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace HerPublicWebsite.Filters;
+
+public class SessionExpiryAttribute : ActionFilterAttribute
+{
+    public override void OnActionExecuting(ActionExecutingContext context)
+    {
+        if (!context.HttpContext.Session.Keys.Any())
```

Comment:
Note that this does not check for the correct session data (e.g. that you have filled in all the previous questions before reaching a particular page), so you might still encounter an error in that case, but that's probably ok - you'd have to be explicitly jumping around the journey.

Reply:
Sounds fine to me